src/components/textarea/textarea.test.tsx

```diff
@@ -0,0 +1,719 @@
+  render(<Textarea inputWidth={30} />);
+  expect(screen.getByTestId("input-presentation-container")).toHaveStyle({
+    flex: "0 0 30%",
+  });
+});
+
+test("when inputWidth is not set but labelWidth is, input width defaults to 100 minus label width", () => {
+  render(<Textarea labelWidth={45} />);
+  expect(screen.getByTestId("input-presentation-container")).toHaveStyle({
+    flex: "0 0 55%",
+  });
+});
+
+test("when an expandable textarea unmounts, the event listener is removed from the window", () => {
+  const removeEventListenerSpy = jest.spyOn(window, "removeEventListener");
+
+  const { unmount } = render(
+    <Textarea
+      value="foo"
+      name="textarea"
+      onChange={jest.fn()}
+      label="Label"
+      expandable
+      rows={10}
+      characterLimit={100}
+    />
+  );
+
+  unmount();
+
+  expect(removeEventListenerSpy).toHaveBeenCalledWith(
+    "resize",
+    expect.any(Function)
+  );
+
+  removeEventListenerSpy.mockRestore();
+});
+
+// TODO: test should be removed once FE-5551 (shared coverage) is implemented. This test is needed for coverage,
+// but it doesn't add value as it still passes even when the bug is reintroduced, as the bug only happens in a
+// real browser environment, not in JSDOM.
+describe("when a parent container scrolls vertically", () => {
+  it("restores the scroll position after expanding", () => {
+    render(
+      <div
+        data-role="scroll-wrapper"
+        style={{ height: "200px", overflowY: "scroll" }}
+      >
+        <div id="inner-wrapper" style={{ height: "1000px" }}>
+          <Textarea
+            value="foo"
+            name="textarea"
+            onChange={jest.fn()}
+            label="Label"
+            expandable
+            rows={10}
+          />
+        </div>
+      </div>
+    );
+
+    const scrollWrapper = screen.getByTestId("scroll-wrapper");
+    const textarea = screen.getByRole("textbox");
+
+    scrollWrapper.scrollTop = 700;
+    jest.spyOn(textarea, "scrollHeight", "get").mockImplementation(() => 500);
+
+    window.dispatchEvent(new Event("resize"));
+
+    expect(scrollWrapper.scrollTop).toBe(700);
+  });
+});
+
+test("when textarea cannot be expanded, it does not remove any event listener from the window", () => {
+  const removeEventListenerSpy = jest.spyOn(window, "removeEventListener");
+
+  const { unmount } = render(
+    <Textarea
+      id="Dummy Area"
+      name="textarea"
+      value="foo"
+      onChange={jest.fn()}
+      label="Label"
+      rows={10}
+    />
+  );
+
+  unmount();
+
+  expect(removeEventListenerSpy).not.toHaveBeenCalledWith(
+    "resize",
+    expect.any(Function)
+  );
+
+  removeEventListenerSpy.mockRestore();
+});
+
+test("the required prop is passed to the input", () => {
+  render(<Textarea required label="required" />);
+  expect(screen.getByRole("textbox")).toBeRequired();
+});
+
+test("when the required prop is set, the label includes the 'required' asterisk", () => {
```

Comment:
Comment: unless this and the `isOptional` one are needed for coverage I'd be happy to remove these as chromatic is capturing them etc